src/main/java/number/Number.java

```diff
@@ -0,0 +1,62 @@
+package number;
+
+import exception.NegativeNumberException;
+import util.StringUtil;
+
+import java.util.Objects;
+
+public class Number {
+
+    public static final Number ZERO = new Number(Number.ZERO_VALUE);
+    public static final int ZERO_VALUE = 0;
+
+    private int value;
+
+    private Number() {
+    }
+
+    private Number(final int number) {
+        validateNegativeNumber(number);
+        this.value = number;
+    }
+
+    private Number(final String number) throws NumberFormatException {
+        this.value = StringUtil.toInt(number);
+        validateNegativeNumber(this.value);
+    }
+
+    public static Number of(final String number) {
+        return new Number(number);
+    }
```

Comment:
```suggestion
    public static Number of(final String number) {
        int parsedInt = StringUtil.toInt(number);
        
        return new Number(parsedInt);
    }
```

이런식으로 하면 중복을 없앨수 있을거 같은데 어떤가요?